test/e2e/helloworld_helm_test.go

```diff
@@ -122,23 +122,22 @@ var _ = ginkgo.Describe("install/uninstall helloworld helm addons", func() {
 			return fmt.Errorf("addon agent deployment %s/%s is not deleted", addonInstallNamespace, agentDeploymentName)
 		}, eventuallyTimeout, eventuallyInterval).ShouldNot(gomega.HaveOccurred())
 
-		gomega.Eventually(func() error {
+		ginkgo.By("clean up the customized agent namespaces")
+		gomega.Eventually(func() bool {
 			_, err := hubKubeClient.CoreV1().Namespaces().Get(context.Background(),
 				agentInstallNamespaceConfig, metav1.GetOptions{})
 			if errors.IsNotFound(err) {
-				return nil
+				return true
 			}
 
-			if err == nil {
-				errd := hubKubeClient.CoreV1().Namespaces().Delete(context.Background(),
-					agentInstallNamespaceConfig, metav1.DeleteOptions{})
-				if errd != nil {
-					return errd
-				}
+			err = hubKubeClient.CoreV1().Namespaces().Delete(context.Background(),
+				agentInstallNamespaceConfig, metav1.DeleteOptions{})
+			if err != nil {
+				return false
 			}
 
-			return err
-		}, eventuallyTimeout, eventuallyInterval).ShouldNot(gomega.HaveOccurred())
+			return false
+		}, eventuallyTimeout, eventuallyInterval).Should(gomega.BeTrue())
```

Comment:
return error is helpful for debugging when it fails since it will print the error message, but return bool will not.